.circleci/config.yml

```diff
@@ -0,0 +1,46 @@
+version: 2.1
+
+commands:
+  install-deps:
+    description: Install dependencies using pre-commit
+    steps:
+      - run:
+          command: |
+            pip install pre-commit tox flit
+            pre-commit install
+          name: Install dependencies
+
+jobs:
+  build:
```

Comment:
This is really "test" since `tox` tests things. But I would just add the `run: tox` step to the deploy steps so deploy is one job. Also maybe call the job "publish" since "deploy" is more infrastructure-y.